packages/typeorm/README.md

```diff
@@ -0,0 +1,145 @@
+# @loopback/typeorm
+
+This module enables TypeORM support in LoopBack.
+
+## Overview
+
+**NOTE**: This module is experimental and evolving.
+
+## Installation
+
+```sh
+npm install --save @loopback/typeorm
+```
+
+## Usage
+
+### Enabling TypeORM support
+
+Enabling TypeORM support requires the use of the `TypeOrmMixin` mixin and the
+`TypeOrmComponent` component. Import them from `@loopback/typeorm` and configure
+your app as shown below.
+
+```ts
+import {TypeOrmComponent, TypeOrmMixin} from '@loopback/typeorm';
+...
+export class AppaApplication extends BootMixin(
+  ServiceMixin(TypeOrmMixin(RestApplication)),
+) {
+  constructor(options: ApplicationConfig = {}) {
+    super(options);
+    ...
+    this.component(TypeOrmComponent);
+    ...
+  }
+}
+
+```
+
+### Creating Entities
+
+[Entities](https://typeorm.io/#/entities) are equivalent to Juggler's Models.
+Define the entities as usual and keep them in a directory named `entities`, or
+even `models` or any name, just make sure to load them correctly in the
+connections.
+
+```ts
+// src/entities/book.entity.ts
+import {Entity, Column, PrimaryColumn} from 'typeorm';
+@Entity()
+export class Photo {
+  @PrimaryColumn()
+  id: number;
+
+  @Column()
+  title: string;
+
+  @Column()
+  isPublished: boolean;
+}
+```
+
+### Creating Connections
+
+[Connections](https://typeorm.io/#/connection) are equivalent to Juggler
+datasources. They contain the connectivity and other details about the
+databases. Define the connection in files with a `.connection.ts` extension and
+keep them in the `connections` directory of the project.
+
+```ts
+// src/connections/sqlite.connection.ts
+import path from 'path';
+import {SqliteConnectionOptions} from '@loopback/typeorm';
+import {Book} from '../entities/';
+
+export const SQLLITE_CONNECTION: SqliteConnectionOptions = {
+  type: 'sqlite',
+  database: path.join(__dirname, 'sqlite-test-db'),
+  entities: [Book],
+  synchronize: true,
+};
+```
+
+Make sure to install the underlying database driver. For example, if you are
+using SQlite, you'll need to install `sqlite3`.
+
+```sh
+npm install sqlite3
+```
+
+Refer to the
+[TypeORM documentation](https://github.com/typeorm/typeorm#installation) for the
+supported databases and the underlying drivers.
+
+### Creating controllers
+
+Controllers continue to work as usual. And you don't have to create repositories
+since TypeORM creates them for you; just inject them in the controllers. The
+repository API is 100% TypeORM
+[repository API](https://typeorm.io/#/repository-api).
+
+```ts
+// src/controllers/book.controller.ts
+import {get, post, Request, requestBody} from '@loopback/rest';
+import {getModelSchemaRef, Repository, typeorm} from '@loopback/typeorm';
```

Comment:
As discussed before, this should be just `getModelSchema`. (No need to use `$ref` schemas anymore, OpenAPI consolidation will take care of de-duplicating schemas.)

IMO, `@loopback/typeorm` should not be coupled with OpenAPI, I think we should either tell users to apply `jsonToSchemaObject` on the schema produced by `@loopback/typeorm`, or else introduce a new package for things that are specific to TypeORM+OpenAPI.

Reply:
> IMO, @loopback/typeorm should not be coupled with OpenAPI,

Need elaboration.

Reply:
`@loopback/openapi-v3` provides things like `@get` and `@api` decorators.

As a user building gRPC or Pub/Sub application using `@loopback/typeorm` for persistence, why do I have to install dependencies for OpenAPI, when I am not using any of that?

Reply:
Also if we add `gRPC` support in the future and implement code to generate gRPC schemas (protocol buffer message type definitions) from TypeORM entities, are we going to add `gRPC` to `@loopback/typeorm` dependencies and thus force REST/OpenAPI users to install gRPC they are not using?

Anyhow, I think this is out of scope of this pull request, but please do create a follow-up task to continue this discussion and rework the code accordingly.

Reply:
Note to self: Create follow up task.